src/Functions/FunctionsLogical.h

```diff
@@ -176,6 +176,8 @@ class FunctionAnyArityLogical : public IFunction
 
     bool useDefaultImplementationForNulls() const override { return !Impl::specialImplementationForNulls(); }
 
+    DataTypePtr getReturnTypeForDefaultImplementationForDynamic() const override { return std::make_shared<DataTypeUInt8>(); }
```

Comment:
The result of a logical function can also be `Nullable(UInt8)` depending on the nullability of its arguments. Isn’t that the case for a dynamic type as well?

Reply:
Can’t a default implementation got something with a `Nullable` type that affects the result type?

Reply:
`getReturnTypeForDefaultImplementationForDynamic` should return non-nullable return type. Dynamic column always can contain nulls, so in default implementation for dynamic we will convert the result type to `Nullable` always.